bazel/protobuf.patch

```diff
@@ -1,29 +1,20 @@
 diff --git a/BUILD.bazel b/BUILD.bazel
-index 0f6e41e3a..c0d2bbccf 100644
+index 69a1ba96c..9b80aa81b 100644
 --- a/BUILD.bazel
 +++ b/BUILD.bazel
-@@ -138,7 +138,7 @@ cc_library(
+@@ -165,14 +165,80 @@ alias(
      visibility = ["//visibility:public"],
-     deps = [":protobuf_lite"] + select({
-         "//build_defs:config_msvc": [],
--        "//conditions:default": ["@zlib//:zlib"],
-+        "//conditions:default": ["//external:zlib"],
-     }),
- )
- 
-@@ -454,14 +454,79 @@ cc_library(
-     deps = [":protobuf"],
  )
  
 +# Envoy: Patch
```

Comment:
@adisuissa just flagging my main motivation here which is to ensure this continues to work

as we dont call the workspace setup function from `rules_proto` we dont get the prebuilt `protoc` binary that protobuf provides - this meant that for a long time all bazel jobs were repeatedly re/building the `protoc` binary from source - this hack prevents that